en/middleware/mavlink.md

```diff
@@ -14,22 +14,28 @@ The tutorial assumes you have a [custom uORB](../middleware/uorb.md) `ca_traject
 
 ## Defining Custom MAVLink Messages
 
+PX4 includes the [mavlink/mavlink](https://github.com/mavlink/mavlink) repo as a submodule under [/src/modules/mavlink](https://github.com/PX4/PX4-Autopilot/tree/master/src/modules/mavlink), and generates the MAVLink 2 C library at build time from [/mavlink/messages/1.0/development.xml](https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/development.xml) (by default).
+The dialect that is built is specified using the variable `MAVLINK_DIALECT` in [/src/modules/mavlink/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/mavlink/CMakeLists.txt#L34).
+
+In order to add your message we recommend that you create your messages in a new dialect file in the same directory, for example `PX4-Autopilot/src/modules/mavlink/mavlink/message_definitions/v1.0/custom_messages.xml`, and set `MAVLINK_DIALECT` to build the new file.
+This dialect file should include `development.xml`.
+
+:::note
+You might alternatively add your messages to `common.xml` or `development.xml`.
+Either way, remember that the same dialect file must be eventually be built in QGroundControl (or whatever software you use to communicate with PX4).
+:::
+
 The MAVLink developer guide explains how to define new messages and build them into new programming-specific libraries:
+
 - [How to Define MAVLink Messages & Enums](https://mavlink.io/en/guide/define_xml_element.html)
-- [Generating MAVLink Libraries](https://mavlink.io/en/getting_started/generate_libraries.html)
+- [Install MAVLink Toolchain](https://mavlink.io/en/getting_started/installation.html)
+- [Generating MAVLink Libraries](https://mavlink.io/en/getting_started/generate_libraries.html).
 
-Your message needs to be generated as a C-library for MAVLink 2.
-Once you've [installed MAVLink](https://mavlink.io/en/getting_started/installation.html) you can do this on the command line using the command:
+Even though PX4 will build the files for you, you should test that your messages are correctly defined by building your XML file on the command line:
 ```sh
-python -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated/include/mavlink/v2.0 message_definitions/v1.0/custom_messages.xml
+python -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated/include/mavlink/v2.0 message_definitions/v1.0/https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/development.xml
```

Comment:
Simpler would be to just check the generated files in the build directory. However I don't think it's typically needed, unless something isn't working and you need to debug it.

Reply:
My thinking was that building in mavlink would simplify debugging if there were problems - for example, understanding how to format messages and seeing the validation errors. I'll rework though.